### DISCUSSION TITLE
Integrating Sirius Web into an application

### DISCUSSION THREAD

**Discussion by mfilipadneto** [eclipse-sirius/sirius-components#3157] [Q&A]
Hello! 
I am Filipa, a computer science student, and I am planning on using Sirius Web for a part of my thesis. 

My idea is to take advantage of Sirius' components in order to build a diagram editor, with a domain and view already defined and restricted (by me). I thought it would be something similar to what is mentioned in [this video](https://www.youtube.com/watch?v=VOH8Pi23vtQ) (minute 22:10). But I could not find any documentation on how to do this.

I already defined my domain and view for my type of diagrams according to Sirius, but I am very unsure on how to actually integrate the graphical part into a separate application.

What I have started doing is adapting the Sirius Web sample application code, available on GitHub, in order to somehow inject my own domain and restrict the user to only be able to use my domain to build a diagram. But I feel that there must be a more correct way to do this.

I hope that I was clear and that you can help!
Thank you

**Comment by mrshaone** [comment]
Hello,
I recently started learning Sirius, and I believe I might have a good solution for you! Why not leverage an iframe in your application to display your own Sirius project in a custom view?
There may be better solutions available from the Sirius team, but I noticed there hasn't been a response to your question, and I just wanted to help. I hope this is useful to you.
